Dark mode

AhmadALshalabi opened the pull request.

AhmadALshalabi committed:
css theme styling v0.1

AhmadALshalabi committed:
add theme toggle + emit data

AhmadALshalabi committed:
add theme toggle

AhmadALshalabi committed:
emit toggletheme class

AhmadALshalabi committed:
improvments

AhmadALshalabi committed:
change to vuex state and enhance styling

AhmadALshalabi committed:
Merge branch 'dev' into Dark-Mode

AhmadALshalabi committed:
fix bug

AhmadALshalabi committed:
Merge branch 'Dark-Mode' of https://github.com/New-BaseGroup/FrontEnd into Dark-Mode

AhmadALshalabi committed:
remove unnecessary style

A-Lord commented on src/components/UI/SideBar.vue:
```diff
@@ -1,63 +1,75 @@
-	{
-		icon: 'notes-medical',
-		title: 'Transaction',
-		link: '/balance',
```
Här vill du skriva över omars nya balance med income och expense

AhmadALshalabi replied:
fixat, rebase funkat inte som man tror

A-Lord commented on src/components/UI/SideBar.vue:
```diff
@@ -1,63 +1,75 @@
-		</button>
-		
-	</div>
+  <div class="flex flex-row relative">
```
Varför har du lagt till relative på sidebaren och varför är det fixed bottom-0 på ThemeToggle .  Det dom gör nu är att förstöra sidan design när man förminskar eller förstorar fönstret med sidan. 
![reactive](https://user-images.githubusercontent.com/90420563/192382372-cae106a8-94f2-43b7-a414-c0d74297b651.png)
![reactive2](https://user-images.githubusercontent.com/90420563/192382375-5e5cc6dd-a9d3-4d31-97d9-0a71efde75ac.png)

A-Lord commented on src/assets/tailwind.css:
```diff
@@ -5,18 +5,48 @@
+    @apply flex  w-full cursor-pointer items-center rounded-l-full pl-3 transition duration-300 ease-in-out ml-1  py-4 my-4   bg-white text-[#3094a1] active:scale-110;
+  }
+  .sideBarButton {
+    @apply absolute top-1/2 left-[100%] mb-auto px-1 py-5  my-auto rounded-r-full hover:text-white hover:bg-[#3094a1] border-[#3094a1] bg-white  text-[#3094a1] border active:scale-110;
```
Här säger du att sidebarknappen ska få gå in och använda sig av maincontents space. med din absolute tag.
![absolute](https://user-images.githubusercontent.com/90420563/192383960-54f83f88-676f-4153-85c9-77f25e7a8b1b.png)

Så här ser det ut i dev just nu,. då håller sig sidebar + knapp bara inom sin flex row. och main content har sin egen flex row . 
![noAboslutPosition](https://user-images.githubusercontent.com/90420563/192383963-8efcf02d-aa42-4295-9372-0ce57ffd5415.png)

AhmadALshalabi committed:
fix balance link

A-Lord reviewed (changes requested):
När du lägger till relative/absolute position måste du se till att sidans design fortfarande funkar reactivt.  Jag rekomederar dock att vara sparsamm med dom då dom inte funkar någe vidare tillsammans med flexbox och grid.    Ta en titt på https://ishadeed.com/article/less-absolute-positioning-modern-css/ t.ex.

AhmadALshalabi committed:
fix styling + active link color

A-Lord committed:
Update App.vue

changed so darkmode div now is forced to be all the content so no content can have wrong mode on.

A-Lord reviewed (approved):
Ser awesome ut! . flyttade så classen som bestämmer om det är dark eller light nu är tvingad till 100% av allt content eller mins hela skärmen så det inte blir så här. 
![scrollbug](https://user-images.githubusercontent.com/90420563/192403941-93fff4a7-4341-472f-8278-aa52c01c475e.png)

AhmadALshalabi committed:
cleaning
